app/services/update_user.rb

```diff
@@ -33,7 +33,24 @@ def update_sam
     reckoner.delay.set!
   end
 
+  def parsed_user_params
+    user_params.each_with_object({ }) do |(key, value), hash|
+      converted_value = nil_not_empty_string(value)
+      hash[key] = converted_value
+    end
+  end
```

Comment:
updated solution: set db default for fields that can be `nil` to `""` so when the form is updated and the value is `""` we do not record that as a change. Removes need to fancy params parsing and is also nice bc now we can render these values in the views without worrying about `nil` errors